packages/calcite-components/src/components/modal/modal.e2e.ts

```diff
@@ -389,38 +391,47 @@ describe("calcite-modal", () => {
     expect(modal).toEqualAttribute("aria-modal", "true");
   });
 
+  const delayInMilliseconds = 300;
+
   it("closes and allows re-opening when Escape key is pressed", async () => {
     const page = await newE2EPage();
-    await page.setContent(`<calcite-modal ></calcite-modal>`);
+    await page.setContent(`<calcite-modal></calcite-modal>`);
     await skipAnimations(page);
     const modal = await page.find("calcite-modal");
-    await modal.setProperty("open", true);
+
+    modal.setProperty("open", true);
     await page.waitForChanges();
     expect(await modal.isVisible()).toBe(true);
+
     await page.keyboard.press("Escape");
     await page.waitForChanges();
     expect(await modal.isVisible()).toBe(false);
     expect(await modal.getProperty("open")).toBe(false);
-    await modal.setProperty("open", true);
+
+    modal.setProperty("open", true);
     await page.waitForChanges();
+    await page.waitForTimeout(delayInMilliseconds);
     expect(await modal.isVisible()).toBe(true);
   });
 
   it("closes when Escape key is pressed and modal is open on page load", async () => {
     const page = await newE2EPage();
-    await page.setContent(`<calcite-modal  open></calcite-modal>`);
+    await page.setContent(`<calcite-modal open></calcite-modal>`);
+    await skipAnimations(page);
+
     const modal = await page.find("calcite-modal");
     await page.waitForChanges();
+    await page.waitForTimeout(delayInMilliseconds);
     expect(modal).toHaveAttribute("open");
-    expect(modal).toHaveAttribute("open");
+
     await page.keyboard.press("Escape");
     await page.waitForChanges();
+    await page.waitForTimeout(delayInMilliseconds);
```

Comment:
maybe `await waitForAnimationFrame()` would work instead?

Reply:
This is the only place where this substitution works.

Reply:
Interesting. I think we could live with the delay but I'm curious on what its waiting for exactly.

Reply:
I had to add them after this commit: [`6bf5a5f` (#9231)](https://github.com/Esri/calcite-design-system/pull/9231/commits/6bf5a5fc531db43d9bb4f94d9cc98157226b08bb)